adapter/internal/oasparser/envoyconf/routes_with_clusters.go

```diff
@@ -853,6 +853,17 @@ func createRoute(params *routeCreateParams) *routev3.Route {
 		if params.rewritePath != "/" {
 			substitutionString = endpointBasepath + params.rewritePath
 		}
+	} else {
+		if strings.Contains(resourcePath, "?") {
+			resourcePath = strings.Split(resourcePath, "?")[0]
+		}
+		resourceRegex := generatePathRegexSegment(resourcePath)
+		pathRegex = "^" + basePath + resourceRegex + "$"
```

Comment:
remove $ as it won't accept query params

Reply:
fixed: https://github.com/wso2/product-microgateway/pull/2837/commits/df5a462739c56f10ff32ab8547bf728f711fc4dc